delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/ResourceResolution.scala

```diff
@@ -0,0 +1,162 @@
+package ch.epfl.bluebrain.nexus.delta.sdk
+
+import cats.implicits._
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.AclAddressFilter.AnyOrganizationAnyProject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.acls.{AclAddress, AclCollection}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.{Caller, Identity}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.permissions.Permission
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.IdentityResolution.{ProvidedIdentities, UseCurrentCaller}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.Resolver.{CrossProjectResolver, InProjectResolver}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResolverResolutionRejection._
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResourceResolutionReport.{ResolverReport, ResolverReportFailed, ResolverReportSuccess}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.{Resolver, ResolverRejection, ResourceResolutionReport}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.SearchParams.ResolverSearchParams
+import ch.epfl.bluebrain.nexus.delta.sdk.model.search.{Pagination, ResultEntry}
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{IdSegment, ResourceRef}
+import monix.bio.{IO, UIO}
+
+/**
+  * Resolution for a given type of resource
+  * @param acls            the acls module
+  * @param resolvers       the resolvers module
+  * @param readPermission  the permission required to get the given type of resource on a project
+  * @param fetchResource   how we can get a resource from a [[ResourceRef]]
+  * @param lensTypes       a lens to extract the types from the given resource
+  */
+final class ResourceResolution[Resource](
+    acls: Acls,
+    resolvers: Resolvers,
+    readPermission: Permission,
+    fetchResource: (ResourceRef, ProjectRef) => IO[FetchRejection, Resource]
+)(implicit lensTypes: Lens[Resource, Set[Iri]]) {
```

Comment:
You don't need this lens if yo change the signature of `fetchResource`  to:
```scala
fetchResource: (ResourceRef, ProjectRef) => IO[FetchRejection, ResourceF[Resource]]
```
since there you have the types too